scripts/components/Dialog/PasswordContent.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { H5PContext } from "../../context/H5PContext";
+import './PasswordContent.scss';
+
+export default class PasswordContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      unlocked: false,
+      hasClicked: false,
+      shakeClass : "",
+      showPasswordInput : true
+    }
+  }
+
+  handleOnClick = event => {
+    event.preventDefault();
+    this.setState({
+      hasClicked: true
+    })
+    if(this.props.currentInteraction.unlocked) {
+      this.props.showInteraction(this.props.currentInteractionIndex)
+    } else {
+      this.setState({
+        unlocked: this.props.handlePassword(this.input.value)
+      })
+      if(!this.props.currentInteraction.unlocked){
+        this.shakeIcon();
+
+      }
+    }
+  };
+  toggleShowCode = () => {
+
+    this.setState({
+      showPasswordInput : !this.state.showPasswordInput
+    })
+
+  }
+
+  shakeIcon = () => {
+    this.setState({
+      shakeClass : "h5p-password-icon--shake"
+    })
+    setTimeout((() => {
+      this.setState({
+        shakeClass : ""
+      })
+    }), 500)
+  };
+  render() {
+//TODO: replace with l10n translation
+    return (
+      <div className='h5p-password-content' >
+        <div className={`h5p-password-icon-wrapper ${this.state.unlocked ? "h5p-password-icon-wrapper--correct-code" :
+          !this.state.hasClicked ? "" : "h5p-password-icon-wrapper--wrong-code"
+        }` }>
+          <span className={`h5p-password-icon ${this.state.unlocked ? "unlocked" : "locked"}` + ` ${this.state.shakeClass}`}/>
+        </div>
+        <h1>{this.state.unlocked ? this.context.l10n.unlocked : this.context.l10n.locked}</h1>
+
+        {<span className={"h5p-field-description"}>
+
+          {this.state.unlocked ? this.context.l10n.contentUnlocked :
+            !this.state.hasClicked ? this.context.l10n.searchRoomForCode : this.context.l10n.wrongCode
+          }
+
+         </span>}
+        <form className={"h5p-wrapper"} onSubmit={this.handleSubmit}>
+          <label className={"h5p-wrapper"} htmlFor={"field-code-" + this.props.currentInteractionIndex}>
+            <span className={"h5p-field-text"}>{this.props.hint}</span>
+            <div className={"h5p-wrapper-inner"}>
+              <input type={this.state.showPasswordInput ? "password" : "text"} className={"h5p-field-input"} id={"field-code-" + this.props.currentInteractionIndex} placeholder={this.context.l10n.code}  ref={(input) => this.input = input}/>
```

Comment:
It's maybe cleaner to have the input value as a state that is set on input change, than using a reference to the input field itself. Also, using ref functions is deprecated in React 16.14, and we should use `React.createRef` instead.

Reply:
Agree, was initally like that, but changed it to ref because of some logic back then didn't work. Tested now with an handling onchange and setting, seems to be working.